### PR TITLE
Stream Indexed Attestations RPC Implementation

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #4836 

---

# Description

**Write why you are making the changes in this pull request**

This PR implements the stream indexed attestations RPC endpoint.

### CODE REVIEW

File: beacon-chain/rpc/beacon/attestations.go [hunk 1]
```diff
@@ -233,7 +246,64 @@ func (bs *Server) StreamAttestations(
...
+				if !ok {
+					// Got bad data over the stream.
+					continue
+				}
+				if data.Attestation == nil {
+					// One nil attestation shouldn't stop the stream.
+					continue
+				}
+				epoch := helpers.SlotToEpoch(bs.HeadFetcher.HeadSlot())
+				committeesBySlot, _, err := bs.retrieveCommitteesForEpoch(stream.Context(), epoch)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/beacon/attestations.go]: Is this cached and/or cheap?

Comment (rauljordan) [reply beacon-chain/rpc/beacon/attestations.go]: there's a cache underneath the hood, it isn't a huge load on slasher but we'll have to do a bit more testing with mainnet/testnet conditions

Comment (terencechain) [reply beacon-chain/rpc/beacon/attestations.go]: this `epoch` can only be current or previous epoch correct? If yes, it's cheap as they are already cached underneath the hood

Comment (rauljordan) [reply beacon-chain/rpc/beacon/attestations.go]: Yep only current!

**Pull request merged**